rust/otap-dataflow/crates/telemetry/src/opentelemetry_client/meter_provider.rs

```diff
@@ -0,0 +1,300 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+//! Configures the OpenTelemetry meter provider based on the provided configuration.
+
+use opentelemetry::global;
+use opentelemetry_otlp::{Protocol, WithExportConfig};
+use opentelemetry_sdk::{
+    Resource,
+    metrics::{MeterProviderBuilder, PeriodicReader, SdkMeterProvider},
+};
+use otap_df_config::pipeline::{
+    MetricsPeriodicExporterConfig, MetricsReaderConfig, OtlpExporterConfig, OtlpProtocol,
+    Temporality,
+};
+
+use crate::error::Error;
+
+/// Wrapper around the OpenTelemetry SDK meter provider and its runtime.
+pub struct MeterProvider {
+    sdk_meter_provider: SdkMeterProvider,
+    runtime: Option<tokio::runtime::Runtime>,
+}
+
+impl MeterProvider {
+    /// Configures the OpenTelemetry meter provider based on the provided configuration.
+    pub fn configure(
+        sdk_resource: Resource,
+        metric_readers: &[MetricsReaderConfig],
+        initial_runtime: Option<tokio::runtime::Runtime>,
+    ) -> Result<MeterProvider, Error> {
+        let mut sdk_meter_builder = SdkMeterProvider::builder();
+        sdk_meter_builder = sdk_meter_builder.with_resource(sdk_resource);
+
+        let mut runtime: Option<tokio::runtime::Runtime> = initial_runtime;
+
+        for reader in metric_readers {
+            (sdk_meter_builder, runtime) =
+                Self::configure_metric_reader(sdk_meter_builder, reader, runtime)?;
+        }
+
+        let sdk_meter_provider = sdk_meter_builder.build();
+
+        global::set_meter_provider(sdk_meter_provider.clone());
+
+        Ok(MeterProvider {
+            sdk_meter_provider,
+            runtime,
+        })
+    }
+
+    /// Consume the MeterProvider and return its components.
+    pub fn deconstruct(self) -> (SdkMeterProvider, Option<tokio::runtime::Runtime>) {
```

Comment:
The Rust idiom for this function is `into_parts()`, I recently learned. ([for example](https://arrow.apache.org/rust/arrow_array/array/struct.PrimitiveArray.html#method.into_parts), we have also `OtapPdata::into_parts` for the pair of Context and Payload).

```suggestion
    pub fn into_parts(self) -> (SdkMeterProvider, Option<tokio::runtime::Runtime>) {
```